xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp

```diff
@@ -1634,6 +1639,11 @@ CDemuxStream* CDVDDemuxFFmpeg::AddStream(int streamIdx)
         st->iOrientation = 0;
         st->iBitsPerPixel = pStream->codecpar->bits_per_coded_sample;
         st->iBitRate = static_cast<int>(pStream->codecpar->bit_rate);
+        st->bitDepth = 8;
+        const AVPixFmtDescriptor* desc =
+            av_pix_fmt_desc_get(static_cast<AVPixelFormat>(pStream->codecpar->format));
+        if (desc != nullptr && desc->comp != nullptr)
+          st->bitDepth = desc->comp[0].depth;
```

Comment:
Should we be using `av_get_bits_per_pixel` here?

Reply:
Yep, seems to do about the same thing - will change.

Reply:
OK the additional computation of this method does not give back what I'm looking for.
For 10 bit AV1 it returns 15, while my approach returns 10 - so I'll leave it as is.

Reply:
What does the mediainfo of this file say? 15 is odd in all combinations.

Reply:
Ah, obviously. It's YUV with a YUV 4:2:0 subsamling. That means: per 4 pixels: 4*10+10+10 = 60 -> 60/4 = 15.

All fine here. You don't want the bits_per_pixel, but the depth I assume.

Reply:
Your original approach was okay I think, also get_bits_depth is going to be deprecated. https://patchwork.ffmpeg.org/project/ffmpeg/patch/20210822145409.316517-2-george@nsup.org/ as well and the fallback is using something similar.